genie-core/src/main/java/com/netflix/genie/core/services/impl/JobCoordinatorServiceImpl.java

```diff
@@ -331,10 +342,81 @@ private void setRuntimeEnvironment(
     private Cluster getCluster(final JobRequest jobRequest) throws GenieException {
         final long start = System.nanoTime();
         try {
-            log.info("Selecting cluster for job {}", jobRequest.getId());
-            final Cluster cluster
-                = this.clusterLoadBalancer.selectCluster(this.clusterService.chooseClusterForJobRequest(jobRequest));
-            log.info("Selected cluster {} for job {}", cluster.getId(), jobRequest.getId());
+            log.info("Selecting cluster for job {}", jobRequest.getId().orElse(NO_ID_FOUND));
+            final List<Cluster> clusters = this.clusterService.chooseClusterForJobRequest(jobRequest);
```

Comment:
Should this list be immutable to avoid plugins messing with it (as of now, I think LBs can add and remove elements).